src/main/kotlin/no/nav/hjelpemidler/soknad/mottak/service/PapirSøknadEndeligJournalført.kt

```diff
@@ -0,0 +1,152 @@
+import no.nav.hjelpemidler.soknad.mottak.metrics.Prometheus
+import java.util.UUID
+
+private val logger = KotlinLogging.logger {}
+private val sikkerlogg = KotlinLogging.logger("tjenestekall")
+
+internal class PapirSøknadEndeligJournalført(rapidsConnection: RapidsConnection, private val store: SøknadStore, private val infotrygdStore: InfotrygdStore) :
+    River.PacketListener {
+    init {
+        River(rapidsConnection).apply {
+            validate { it.demandValue("eventName", "PapirSoeknadEndeligJournalfoert") }
+            validate {
+                it.requireKey(
+                    "fodselNrBruker",
+                    "hendelse",
+                    "hendelse.journalingEvent",
+                    "hendelse.journalingEvent.journalpostId",
+                    "eventId",
+                    "hendelse.journalingEventSAF",
+                    "hendelse.journalingEventSAF.sak",
+                    "hendelse.journalingEventSAF.sak.fagsakId"
+                )
+            }
+        }.register(this)
+    }
+
+    private val JsonMessage.eventId get() = this["eventId"].textValue()
+    private val JsonMessage.fnrBruker get() = this["fodselNrBruker"].textValue()
+    private val JsonMessage.journalpostId get() = this["hendelse"]["journalingEvent"]["journalpostId"].asInt()
+    private val JsonMessage.fagsakId get() = this["hendelse"]["journalingEventSAF"]["sak"]["fagsakId"].textValue()
+
+    override fun onPacket(packet: JsonMessage, context: RapidsConnection.MessageContext) {
+        runBlocking {
+            withContext(Dispatchers.IO) {
+                launch {
+
+                    if (skipEvent(UUID.fromString(packet.eventId))) {
+                        logger.info { "Hopper over event i skip-list: ${packet.eventId}" }
+                        return@launch
+                    }
+
+                    val soknadId = UUID.randomUUID()
+                    val fnrBruker = packet.fnrBruker
+                    val fagsakId = packet.fagsakId
+
+                    val vedtaksresultatData = VedtaksresultatData(
+                        soknadId,
+                        fnrBruker,
+                        VedtaksresultatData.getTrygdekontorNrFromFagsakId(fagsakId),
+                        VedtaksresultatData.getSaksblokkFromFagsakId(fagsakId),
+                        VedtaksresultatData.getSaksnrFromFagsakId(fagsakId),
+                    )
+
+                    try {
+                        val soknadData = PapirSøknadData(
+                            fnrBruker = fnrBruker,
+                            soknadId = soknadId,
+                            status = Status.ENDELIG_JOURNALFØRT,
+                            journalpostid = packet.journalpostId
+                        )
+
+                        if (store.soknadFinnes(soknadData.soknadId)) {
+                            logger.warn { "En søknad med denne id-en er allerede lagret i databasen: $soknadId" }
+                            return@launch
```

Comment:
Blir eventen hoppa over i det stille dersom vi havnar her? Burde det kastast exception i staden, slik at vi får retry?

Reply:
Har endret denne til å sjekke for fnr og journalpostid. Men den skal egentlig bare sjekke om en søknad med lik fnr og journalpostid allerede er lagret i databasen. Hvis dette finnes har det nok skjedd en feil et sted, så tror det er greit at den bare logger og returnerer uten retry.